[Benjamin Tan] iP

BenjaminTan99 opened the pull request:
# _Welcome to Ben's!_

> Your most unhappy customers are your greatest source of learning. - Bill Gates

Ben's is a simple, interactive application designed to:

    - Keep track of tasks 📆 
    - Retrieve your tasks, whether it is in the past, present or future! 
    - Change and update tasks anytime, anywhere!

# **At Ben's, the customer is always right. 😄** 
Feel free to comment and give feedback on Ben's! We welcome any customers' feedback as we strive to improve ourselves constantly.

# Features
1. Add or remove tasks and mark them as done.
2. Unique types of tasks (currently limited to 3):     
      - [x] To-Dos
      - [x] Deadlines
      - [x] Events
3. Search and locate tasks via keywords
4. Actively 'replying' UI! (i.e. few system errors causing Ben's to crash, allowing continuous input)
5. Ben's is completely **FREE**! No charges or fees required to start.

# How to use
1. Install the .jar file [here](https://github.com/BenjaminTan99/ip/releases/tag/v0.1).
2. Run the .jar file (you may learn how to open .jar files [here](https://se-education.org/guides/tutorials/jar.html).
3. Enjoy your time at Ben's! Remember to give feedback!


What you should see for the main class:
```java
public static void main(String[] args) {
        new Duke("data/duke.txt").run();
}
```

Commit:
Add Gradle support

BenjaminTan99 committed:
Added Level-1

BenjaminTan99 committed:
Added Level-2

BenjaminTan99 committed:
Revised Level-2

BenjaminTan99 committed:
Added Level-3 with new class Task

BenjaminTan99 committed:
Added Level-4 along with Classes Deadline, Event and ToDos

BenjaminTan99 committed:
changed with errors

BenjaminTan99 committed:
improved Level-4

BenjaminTan99 committed:
minor edits

BenjaminTan99 committed:
Added Level-5

BenjaminTan99 committed:
Added Level-6

BenjaminTan99 committed:
Added Level-7 and documentation

BenjaminTan99 committed:
Added Level-8

BenjaminTan99 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Resolution: Keep both

BenjaminTan99 committed:
Added A-MoreOOP

BenjaminTan99 committed:
Added A-Packages

BenjaminTan99 committed:
Added A-JUnit

BenjaminTan99 committed:
Added A-Jar

BenjaminTan99 committed:
Added A-JavaDoc

BenjaminTan99 committed:
Added A-CodingStandard

BenjaminTan99 committed:
Added Level-9

BenjaminTan99 committed:
small update

BenjaminTan99 committed:
Revert "small update"

This reverts commit 99bce59dd81e1f09fef552392016c213f6f33ea5.

BenjaminTan99 committed:
minor changes

BenjaminTan99 committed:
minor changes

BenjaminTan99 committed:
Merge branch 'branch-Level-9'

BenjaminTan99 committed:
minor changes to Duke

BenjaminTan99 committed:
Minor edits for JavaDocs

BenjaminTan99 committed:
Minor edits for JavaDocs

BenjaminTan99 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

BenjaminTan99 committed:
Added A-Gradle

BenjaminTan99 committed:
Merge branch 'branch-Level-8' into branch-A-JavaDoc

# Conflicts:
#	data/duke.txt

BenjaminTan99 committed:
Added gradle

BenjaminTan99 committed:
Minor changes to saved file

BenjaminTan99 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat
Deleted new changes

lingshanng commented on src/main/java/duke/ToDos.java:
```diff
@@ -0,0 +1,55 @@
+ * This class simulates a unique type of Task which only has description
+ * but no designated deadline.
+ */
+public class ToDos extends Task {
```
instead of `ToDos`, maybe `ToDo` will be better to represent a single object

lingshanng commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * The goodbye message displayed after user exits Duke (i.e. bye).
```
maybe can phrase this similar to the others. `Displays the goodbye message...`

BenjaminTan99 committed:
Minor changes

lingshanng reviewed (commented):
Looks good overall!

BenjaminTan99 committed:
Added A-CheckStyle

BenjaminTan99 committed:
Added Parts 1-3 of GUI

luffingluffy commented on src/main/java/duke/DateTimeConverter.java:
```diff
@@ -0,0 +1,56 @@
+        int actualTime = Integer.parseInt(time);
+        String hour;
+        String minute;
+        if (actualTime >= 1200) {
```
I think you can try using switch statement instead of if-elses.

luffingluffy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,194 @@
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/Baba.jpg"));
+//    /** The storage class used to store files for Duke. **/
```
I don't think you should leave this here?

luffingluffy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,194 @@
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
```
I think you should remove unnecessary comments.

luffingluffy reviewed (commented):
Other than some minor fixing, I think it looks good!

BenjaminTan99 committed:
Added Level-10

BenjaminTan99 committed:
Fixed minor issues with Duke's starting messages

BenjaminTan99 committed:
Fixed minor issues

zzybluebell commented on src/main/java/duke/Ui.java:
```diff
@@ -132,6 +132,30 @@ public static void invalidTask() {
+
     /**
      * Displays a message indicating that contents have been written safely and correctly
      * to the designated file.
```
Overall looks very good

zzybluebell commented on src/main/java/Duke.java:
```diff
@@ -1,147 +1,45 @@
-            userInput = newScan.nextLine();
-        }
-        int count = 0;
+        Ui.importantMessage();
```
Very good looking

zzybluebell commented on src/main/java/duke/ToDos.java:
```diff
@@ -7,7 +7,7 @@
 public class ToDos extends Task {
```
Perhaps here should be named as Todo.java

BenjaminTan99 committed:
Add Assertions

Needed to verify the validity of input variables.

BenjaminTan99 committed:
Merge pull request #1 from BenjaminTan99/branch-A-Assertions

Add Assertions

BenjaminTan99 committed:
Improve Code Quality

Changed to meet Standard Coding Qualities.

Added constants, reduced length of nested code and other changes.

To improve readability of code and simplify it.

BenjaminTan99 committed:
Merge branch 'master' into branch-A-CodeQuality

BenjaminTan99 committed:
Merge pull request #2 from BenjaminTan99/branch-A-CodeQuality

Improve Code Quality

BenjaminTan99 committed:
Add C-FriendlierSyntax and enhanced code quality.

Simplify inputs/actions for expert/advanced users.

Provided the option of shortforms like del for delete and others as a command.

Ensure that even with shortforms, the command is still understandable.

BenjaminTan99 commented:
Accidental closure of pull request, reopening

BenjaminTan99 committed:
Minor fixes with regards to long methods

BenjaminTan99 committed:
Fixed sizing issues with Duke

BenjaminTan99 committed:
Modify to if-else statement

BenjaminTan99 committed:
Fixed issues with welcome message not displaying

BenjaminTan99 committed:
Added A-UserGuide and screenshot

BenjaminTan99 committed:
Set theme jekyll-theme-cayman

BenjaminTan99 committed:
Add link to README

BenjaminTan99 committed:
Merge branch 'master' of https://github.com/BenjaminTan99/ip

BenjaminTan99 committed:
Fix checkstyle issues

BenjaminTan99 committed:
Fix minor issues with checkstyle

BenjaminTan99 committed:
Fix issues with addressing and directories of files

BenjaminTan99 committed:
Simplify some code logic and resetted duke.txt

BenjaminTan99 committed:
Minor fixes for final JAR file

BenjaminTan99 committed:
Fix minor issue when loading empty storage
